internal/txs-engine/txs_engine.go

```diff
@@ -0,0 +1,253 @@
+package txsengine
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/darchlabs/synchronizer-v2"
+	transactionstorage "github.com/darchlabs/synchronizer-v2/internal/storage/transaction"
+	"github.com/darchlabs/synchronizer-v2/pkg/smartcontract"
+	"github.com/darchlabs/synchronizer-v2/pkg/transaction"
+	"github.com/darchlabs/synchronizer-v2/pkg/util"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type idGenerator func() string
+
+type TxsEngine interface {
+	Start(seconds int64)
+	Run() error
+	Halt()
+	GetStatus() StatusEngine
+	SetStatus(status StatusEngine)
+}
+
+type T struct {
+	ScStorage               synchronizer.SmartContractStorage
+	transactionStorage      *transactionstorage.Storage
+	Status                  StatusEngine
+	idGen                   idGenerator
+	NetworksEtherscanURL    map[string]string
+	NetworksEtherscanAPIKey map[string]string
+	NetworksNodesURL        map[string]string
+}
+
+// Define the enigne status
+type StatusEngine string
+
+const (
+	StatusIdle     StatusEngine = "idle"
+	StatusRunning  StatusEngine = "running"
+	StatusStopping StatusEngine = "stopping"
+	StatusStopped  StatusEngine = "stopped"
+	StatusError    StatusEngine = "error"
+)
+
+func New(ss synchronizer.SmartContractStorage, ts *transactionstorage.Storage, idGen idGenerator, etherscanUrlMap map[string]string, etherscanApiKeyMap map[string]string, nodesUrlMap map[string]string) *T {
```

Comment:
the input is too long, you should use a structure as input, also you do not validate if they are correctly defined

Reply:
Ok

Reply:
Done